ios/keyman/Keyman/Keyman/MainViewController.swift

```diff
@@ -879,51 +878,44 @@ class MainViewController: UIViewController, TextViewDelegate, UIActionSheetDeleg
     }
 
     let params = self.params(of: query)
-    if params["url"] == nil {
+    if let urlString = params["url"] {
       // Download and set custom keyboard
-
-      let jsonUrl = params["url"].flatMap { URL(string: $0) }
-      let direct = params["direct"]
-
-      if let jsonUrl = jsonUrl {
-        Manager.shared.dismissKeyboardPicker(self)
-        if !infoView.view.isHidden {
-          perform(#selector(self.infoButtonClick), with: nil)
-        }
-
-        let isDirect = direct == "true"
-
-        customKeyboardToDownload = [
-          "url": jsonUrl,
-          "direct": isDirect
-        ]
-        let title = "Custom Keyboard: \(jsonUrl.lastPathComponent)"
-        showAlert(withTitle: title, message: "Would you like to install this keyboard?",
-                  cancelButtonTitle: "Cancel", otherButtonTitles: "Install", tag: 2)
-      } else {
+      guard let url = URL(string: urlString) else {
         showAlert(withTitle: "Custom Keyboard", message: "The keyboard could not be installed: Invalid Url",
                   cancelButtonTitle: "OK", otherButtonTitles: nil, tag: -1)
         launchUrl = nil
+        return
       }
-    } else {
+
+      Manager.shared.dismissKeyboardPicker(self)
+      if !infoView.view.isHidden {
+        perform(#selector(self.infoButtonClick), with: nil)
+      }
+
+      customKeyboardToDownload = [
+        "url": url,
+        "direct": params["direct"] == "true"
```

Comment:
Let's ignore the 'direct' parameter and always go with direct. Deprecates that additional layer of complexity.